File: MATTHEW/leetcode/AIcoding/3.DeepSeek-v3/javascript_solutions/2269-find-the-k-beauty-of-a-number.js
var divisorSubstrings = function(num, k) {
    const numStr = num.toString();
    let count = 0;
    for (let i = 0; i <= numStr.length - k; i++) {
        const substring = numStr.substring(i, i + k);
        const subNum = parseInt(substring, 10);
        if (subNum !== 0 && num % subNum === 0) {
            count++;
        }
    }
    return count;
};

File: MATTHEW/leetcode/AIcoding/3.DeepSeek-v3/erlang_solutions/2269-find-the-k-beauty-of-a-number.erl
-spec divisor_substrings(Num :: integer(), K :: integer()) -> integer().
divisor_substrings(Num, K) ->
    S = integer_to_list(Num),
    Len = length(S),
    Count = 0,
    if 
        K > Len -> 0;
        true -> count_divisors(S, K, 0, 0, Len)
    end.

count_divisors(S, K, Index, Count, Len) when Index + K =< Len ->
    SubStr = string:substr(S, Index + 1, K),
    SubNum = list_to_integer(SubStr),
    if 
        SubNum =/= 0 andalso (Len =:= K orelse string:substr(S, Index + 1, 1) =/= "0") ->
            case list_to_integer(S) rem SubNum of
                0 -> count_divisors(S, K, Index + 1, Count + 1, Len);
                _ -> count_divisors(S, K, Index + 1, Count, Len)
            end;
        true -> count_divisors(S, K, Index + 1, Count, Len)
    end;
count_divisors(_, _, _, Count, _) -> Count.